envoy/type/percent.proto

```diff
@@ -0,0 +1,47 @@
+syntax = "proto3";
+
+package envoy.type;
+
+import "google/protobuf/wrappers.proto";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Percent]
+
+// Identifies a percentage, in the range [0.0, 100.0].
+message Percent {
+  double value = 1 [(validate.rules).double = {gte: 0, lte: 100}];
+}
+
+// A fractional percentage is used in cases in which for performance reasons performing floating
+// point to integer conversions during randomness calculations is undesirable. The message includes
+// both a numerator and denominator that together determine the final fractional value.
+//
+// * **Example**: 1/100 = 1%.
+// * **Example**: 3/10000 = 0.03%.
+message FractionalPercent {
+  // Specifies the numerator. Defaults to 0.
+  uint32 numerator = 1;
+
+  // Fraction percentages support several fixed denominator values.
+  enum DenominatorType {
+    // 100.
+    //
+    // **Example**: 1/100 = 1%.
+    HUNDRED = 0;
+
+    // 10,000.
+    //
+    // **Example**: 1/10000 = 0.01%.
+    TEN_THOUSAND = 1;
+
+    // 1,000,000.
+    //
+    // **Example**: 1/1000000 = 0.0001%.
+    MILLION = 2;
```

Comment:
Would it theoretically be faster to make these powers-of-two, that way you can perform operations with just shift? Is using decimal preferred because it's simpler for folks to reason about and the speedup vs. integer multi/div negligible?

Reply:
Compilers/processors do a pretty good job of division by a constant, and this is easier for humans to grok, so I'd say leave it how it is.

Reply:
The reality is that we always do modulo with this value followed by comparison. We never do actual division. Either way, I think we should keep decimal it's vastly easier to understand.